dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -42,12 +43,18 @@
 import java.io.OutputStream;
 import java.lang.reflect.Type;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+/**
+ * items:
+ * - we don't want to use `String` for the error identifier. Let's create an `ErrorName` class.
```

Comment:
`ErrorType#name` is sent over the wire, which is a string. It seems fine to keep this as a string.